conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosGovernanceOperations.scala

```diff
@@ -0,0 +1,279 @@
+package tech.cryptonomic.conseil.indexer.tezos
+
+import tech.cryptonomic.conseil.common.tezos.TezosTypes._
+import tech.cryptonomic.conseil.common.tezos.VotingOperations._
+import scala.concurrent.{ExecutionContext, Future}
+import com.typesafe.scalalogging.LazyLogging
+import slick.jdbc.PostgresProfile.api.Database
+import cats.implicits._
+
+/** Process blocks and voting data to compute details for
+  * the governance-related cycles
+  */
+object TezosGovernanceOperations extends LazyLogging {
+
+  /** Collects all relevant data for the governance
+    * voting process
+    *
+    * @param hash identifies a specific block
+    * @param metadata the block metadata
+    * @param proposalId a specific proposal under evaluation
+    * @param allRolls ballot rolls up to this block for the proposal
+    * @param rollsPerLevel ballot rolls included in this specific block
+    * @param ballotsPerCycle how many baloots for a whole cycle
+    * @param ballotsPerLevel how many ballots for the single block level
+    */
+  case class GovernanceAggregate(
+      hash: BlockHash,
+      metadata: BlockHeaderMetadata,
+      proposalId: Option[ProtocolId],
+      allRolls: VoteRollsCounts,
+      rollsPerLevel: VoteRollsCounts,
+      ballotsPerCycle: Option[Voting.BallotCounts],
+      ballotsPerLevel: Option[Voting.BallotCounts]
+  )
+
+  //wrappings to make the distinction between two different counts
+  private case class BallotsPerCycle(counts: Voting.BallotCounts) extends AnyVal
+  private case class BallotsPerLevel(counts: Voting.BallotCounts) extends AnyVal
+
+  /** Collects rolls of votes counted during ballots
+    *
+    * @param yay how many yays
+    * @param nay how many nays
+    * @param pass how many passes
+    */
+  case class VoteRollsCounts(yay: Int, nay: Int, pass: Int)
+
+  /** Extracts and aggregates the information relative to different
+    * governance periods, restricted to a selection of blocks.
+    * We use the input voting rolls and additional data retrieved both
+    * from the tezos node, for the currently examined blocks, and from
+    * previously indexed voting data entries, extracted from db.
+    * The input keys identifies all blocks we're interested in, independently
+    * from having any actual rolls data in the associated map value.
+    *
+    * @param db the reference database
+    * @param node the operator to get information from the remore tezos node
+    * @param bakerRollsByBlock blocks of interest, with any rolls data available
+    * @return the computed aggregate data
+    */
+  def extractGovernanceAggregations(
+      db: Database,
+      nodeOperator: TezosNodeOperator
+  )(
+      bakerRollsByBlock: Map[Block, List[Voting.BakerRolls]]
+  )(implicit ec: ExecutionContext): Future[List[GovernanceAggregate]] = {
```

Comment:
Passing around the execution context is annoying but I think the code structure is much better now! :+1: